From a607fde1df302bf38114790c52eddacbb2f8b100 Mon Sep 17 00:00:00 2001
From: KunYi Chen <kunyi.chen@gmail.com>
Date: Thu, 17 Nov 2022 17:40:59 +0800
Subject: [PATCH 2/5] update DEPANDS of pyinstaller

---
 .../python/python3-nuwriter_0.90.0.bb                 | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/meta-ma35d1/recipes-devtools/python/python3-nuwriter_0.90.0.bb b/meta-ma35d1/recipes-devtools/python/python3-nuwriter_0.90.0.bb
index b2bb8b7..701dbfc 100644
--- a/meta-ma35d1/recipes-devtools/python/python3-nuwriter_0.90.0.bb
+++ b/meta-ma35d1/recipes-devtools/python/python3-nuwriter_0.90.0.bb
@@ -23,8 +23,9 @@ SRC_URI += " file://header-nand.json \
 
 DEPENDS += " \
     libusb1-native \
-    pyinstaller-native \
-    pyinstaller-hooks-contrib-native \
+    python3-native \
+    python3-pyinstaller-native \
+    python3-pyinstaller-hooks-contrib-native \
     python3-altgraph-native \
     python3-pyusb-native \
     python3-pycryptodome-native \
@@ -52,8 +53,8 @@ do_install(){
     cp ${WORKDIR}/pack-nand.json  ${D}${datadir}/nuwriter/
     cp ${WORKDIR}/pack-spinand.json  ${D}${datadir}/nuwriter/
     cp ${WORKDIR}/pack-sdcard.json  ${D}${datadir}/nuwriter/
-   
-    cp ${S}/ddrimg/* ${D}${datadir}/nuwriter/ddrimg/ 
+
+    cp ${S}/ddrimg/* ${D}${datadir}/nuwriter/ddrimg/
     cp ${S}/xusb.bin ${D}${datadir}/nuwriter/
 }
 
@@ -61,7 +62,7 @@ do_deploy() {
     install -d ${DEPLOYDIR}/${BOOT_TOOLS}/nuwriter
     install -d ${DEPLOYDIR}/${BOOT_TOOLS}/nuwriter/ddrimg
     cp -rf ${B}/nuwriter/* ${DEPLOYDIR}/${BOOT_TOOLS}/nuwriter
-    
+
     cp ${S}/ddrimg/* ${DEPLOYDIR}/${BOOT_TOOLS}/nuwriter/ddrimg/
     cp ${S}/xusb.bin  ${DEPLOYDIR}/${BOOT_TOOLS}/nuwriter/
     cp ${S}/xusb.bin  ${DEPLOYDIR}/${BOOT_TOOLS}/
-- 
2.38.1

